core/src/main/java/org/apache/iceberg/rest/ParserContext.java

```diff
@@ -44,7 +44,7 @@ static Builder builder() {
   }
 
   static class Builder {
-    private Map<String, Object> data;
+    private final Map<String, Object> data;
```

Comment:
wouldn't we doing a put on a final object then ?

Reply:
the `final` applies to the actual map instance and doesn't imply that the map is immutable. Looking at the code again I don't think it actually works currently when the builder's `add()` method is called because `data` is an empty immutable map. @singhpk234 shouldn't the `data` map rather be just an empty mutable map?

```
-    private final Map<String, Object> data;
-
-    private Builder() {
-      this.data = Collections.emptyMap();
-    }
+    private final Map<String, Object> data = Maps.newHashMap();
```

Reply:
Yeah good eye, I think we would've caught this when implementing the actual client side changes for scan planning w tests, we wouldn't have any of the expected context passed through but it's good to catch this now.